src/trunk.c

```diff
@@ -7092,23 +7229,57 @@ trunk_range(trunk_handle  *spl,
    return rc;
 }
 
+/*
+ * Allocate memory for stats-related structures off the trunk.
+ */
+static trunk_stats *
+trunk_stats_init(trunk_handle *spl)
+{
+   platform_memfrag memfrag;
+   spl->stats =
+      TYPED_ARRAY_ZALLOC_MF(spl->heap_id, spl->stats, MAX_THREADS, &memfrag);
+   platform_assert(spl->stats);
+
+   spl->stats_size = memfrag_size(&memfrag);
+   for (uint64 i = 0; i < MAX_THREADS; i++) {
+      platform_status rc;
+      rc = platform_histo_create(spl->heap_id,
+                                 LATENCYHISTO_SIZE + 1,
+                                 latency_histo_buckets,
+                                 &spl->stats[i].insert_latency_histo);
+      platform_assert_status_ok(rc);
+      rc = platform_histo_create(spl->heap_id,
+                                 LATENCYHISTO_SIZE + 1,
+                                 latency_histo_buckets,
+                                 &spl->stats[i].update_latency_histo);
+      platform_assert_status_ok(rc);
+      rc = platform_histo_create(spl->heap_id,
+                                 LATENCYHISTO_SIZE + 1,
+                                 latency_histo_buckets,
+                                 &spl->stats[i].delete_latency_histo);
+      platform_assert_status_ok(rc);
+   }
+   return spl->stats;
+}
 
 /*
- *-----------------------------------------------------------------------------
- * Create/destroy
- * XXX Fix this api to return platform_status
- *-----------------------------------------------------------------------------
+ * Allocate memory for trunk handle and do initialization that is common between
+ * 'create' and 'mount' operations.
  */
-trunk_handle *
-trunk_create(trunk_config     *cfg,
-             allocator        *al,
-             cache            *cc,
-             task_system      *ts,
-             allocator_root_id id,
-             platform_heap_id  hid)
+static trunk_handle *
+trunk_handle_init(trunk_config     *cfg,
```

Comment:
Modularized common code from `trunk_create` and `trunk_mount`, so that the memory-related changes could be applied to just one place.